tests/Fixtures/TestBundle/State/Issue5605/MainResourceProvider.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Fixtures\TestBundle\State\Issue5605;
+
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use ApiPlatform\Tests\Fixtures\TestBundle\ApiResource\Issue5605\MainResource;
+use ApiPlatform\Tests\Fixtures\TestBundle\ApiResource\Issue5605\SubResource;
+use ApiPlatform\Tests\Fixtures\TestBundle\Entity\DummyWithSubEntity;
+
+class MainResourceProvider implements ProviderInterface
```

Comment:
do you really need this? isn't the default provider enough to test?

Reply:
I got used to always add custom providers when i use entityClass, but i'll check if it's actually useful here.

Reply:
Just checked - as is if i remove the custom provider i get "Can't get a way to read the property "subResource" in class ...".

And if i rename $subResource to $subEntity so it matches the entities, i get "No resource class found for object of type "ApiPlatform\Tests\Fixtures\TestBundle\Entity\DummySubEntity"."

I don't see an easy way to do this without custom providers.